### PR TITLE
Merge updates from support-spark3.0

### PR DESCRIPTION
**Pull request by firestarman**
This PR is to merge the latest updates from internal branch "support-spark3.0" to support Spark3.0.

### CODE REVIEW

File: datasets/ETL/MortgageETL.ipynb [hunk 1]
```diff
@@ -1,439 +1,1589 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequirement\n",
+    "### 1. Download data\n",
+    "All data could be found at https://rapidsai.github.io/demos/datasets/mortgage-data\n",
+    "\n",
+    "### 2. Download needed jars\n",
+    "* [cudf-0.14-cuda10.1.jar](https://repo1.maven.org/maven2/ai/rapids/cudf/0.14/) (Please choose the correct cudf based one your cuda version, here we use cuda10.1)\n",
+    "* [rapids-4-spark_2.12-0.1.jar](https://urm.nvidia.com/artifactory/sw-spark-maven-local/ai/rapids/rapids-4-spark_2.12/0.1-SNAPSHOT/)\n",
```
Comment (GaryShen2008) [inline datasets/ETL/MortgageETL.ipynb]: Should change to maven central link and the jar version is wrong.

File: datasets/ETL/MortgageETL.ipynb [hunk 2]
```diff
@@ -1,439 +1,1589 @@
...
+    "\n",
+    "\n",
+    "### 3. Start Spark Standalone\n",
+    "Before Running the script, please setup Spark standalone mode\n",
+    "\n",
+    "### 4. Add ENV\n",
+    "```\n",
+    "$ export SPARK_JARS=cudf-0.14-cuda10.1.jar,rapids-4-spark_2.12-0.1.jar\n",
```
Comment (GaryShen2008) [inline datasets/ETL/MortgageETL.ipynb]: rapids-4-spark_2.12-0.1.0.jar

File: examples/apps/scala/pom.xml [hunk 3]
```diff
@@ -14,24 +14,24 @@
...
+        <cuda.classifier>cuda10-1</cuda.classifier>
     </properties>
 
     <repositories>
         <repository>
-            <id>spark.sw.nvidia.com</id>
+            <id>snapshots</id>
             <name>sw-spark-maven</name>
-            <url>https://gpuwa.nvidia.com/artifactory/sw-spark-maven</url>
+            <url>https://urm.nvidia.com:443/artifactory/sw-spark-maven</url>
```
Comment (GaryShen2008) [inline examples/apps/scala/pom.xml]: Should we remove this urm and use maven central for customer building?

File: examples/notebooks/python/agaricus-gpu.ipynb [hunk 4]
```diff
@@ -201,22 +200,22 @@
...
-      "|  1.0|[-0.9667757749557...|[0.03322422504425...|       1.0|\n",
-      "|  0.0|[-0.0080436170101...|[0.99195638298988...|       0.0|\n",
-      "|  0.0|[-0.0080436170101...|[0.99195638298988...|       0.0|\n",
-      "|  0.0|[-0.1416745483875...|[0.85832545161247...|       0.0|\n",
-      "|  0.0|[-0.0747678577899...|[0.92523214221000...|       0.0|\n",
+      "|  1.0|[-0.5428417325019...|[0.45715826749801...|       1.0|\n",
+      "|  0.0|[-0.4572062194347...|[0.54279378056526...|       0.0|\n",
+      "|  0.0|[-0.4572062194347...|[0.54279378056526...|       0.0|\n",
+      "|  0.0|[-0.4514296054840...|[0.54857039451599...|       0.0|\n",
+      "|  0.0|[-0.5428417325019...|[0.45715826749801...|       1.0|\n",
```
Comment (GaryShen2008) [inline examples/notebooks/python/agaricus-gpu.ipynb]: I don't think we need to update this file.
Seems the difference is output with a worse prediction? Maybe the train dataset is too small?

File: examples/notebooks/python/mortgage-gpu.ipynb [hunk 5]
```diff
@@ -112,8 +112,8 @@
     "])\n",
     "features = [ x.name for x in schema if x.name != label ]\n",
     "\n",
-    "train_data = reader.schema(schema).option('header', True).csv('/data/mortgage/csv/train')\n",
-    "trans_data = reader.schema(schema).option('header', True).csv('/data/mortgage/csv/test')"
+    "train_data = reader.schema(schema).option('header', True).csv('/data/mortgage/csv/small-train.csv')\n",
```
Comment (GaryShen2008) [inline examples/notebooks/python/mortgage-gpu.ipynb]: Also, I don't think we need to update this file.
Nothing changed but output.

File: examples/notebooks/scala/mortgage_gpu_crossvalidation.ipynb [hunk 6]
```diff
@@ -0,0 +1,527 @@
...
+    "\n",
+    "NOTE: in this notebook, we have uploaded dependency jars when installing toree kernel. If we don't upload them at installation time, we can also upload in notebook by [%AddJar magic](https://toree.incubator.apache.org/docs/current/user/faq/). However, there's one restriction for `%AddJar`: the jar uploaded can only be available when `AddJar` is called after a new spark session is created. We must use it as below:\n",
+    "\n",
+    "```scala\n",
+    "import org.apache.spark.sql.SparkSession\n",
+    "val spark = SparkSession.builder().appName(\"Taxi-GPU\").getOrCreate\n",
+    "%AddJar file:/data/libs/cudf-0.14-SNAPSHOT-cuda10.jar\n",
+    "%AddJar file:/data/libs/xgboost4j_3.0-1.0.0-SNAPSHOT.jar\n",
+    "%AddJar file:/data/libs/xgboost4j-spark_3.0-1.0.0-SNAPSHOT.jar\n",
+    "%AddJar file:/data/libs/rapids-4-spark-0.1-SNAPSHOT.jar\n",
```
Comment (GaryShen2008) [inline examples/notebooks/scala/mortgage_gpu_crossvalidation.ipynb]: Should update these jar files.

File: examples/notebooks/scala/taxi_gpu_crossvalidation.ipynb [hunk 7]
```diff
@@ -0,0 +1,508 @@
...
+    "import org.apache.spark.ml.tuning.ParamGridBuilder\n",
+    "import org.apache.spark.sql.SparkSession\n",
+    "import org.apache.spark.sql.types.{FloatType, IntegerType, StructField, StructType}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "what is new to xgboost-spark users is rapids.GpuDataReader and **rapids.CrossValidator**"
```
Comment (GaryShen2008) [inline examples/notebooks/scala/taxi_gpu_crossvalidation.ipynb:32]: We don't have GpuDataReader. Seems it spark 2.x code.

File: examples/notebooks/scala/taxi_gpu_crossvalidation.ipynb [hunk 8]
```diff
@@ -0,0 +1,508 @@
...
+    "\n",
+    "NOTE: in this notebook, we have uploaded dependency jars when installing toree kernel. If we don't upload them at installation time, we can also upload in notebook by [%AddJar magic](https://toree.incubator.apache.org/docs/current/user/faq/). However, there's one restriction for `%AddJar`: the jar uploaded can only be available when `AddJar` is called after a new spark session is created. We must use it as below:\n",
+    "\n",
+    "```scala\n",
+    "import org.apache.spark.sql.SparkSession\n",
+    "val spark = SparkSession.builder().appName(\"Taxi-GPU\").getOrCreate\n",
+    "%AddJar file:/data/libs/cudf-0.14-SNAPSHOT-cuda10.jar\n",
+    "%AddJar file:/data/libs/xgboost4j_3.0-1.0.0-SNAPSHOT.jar\n",
+    "%AddJar file:/data/libs/xgboost4j-spark_3.0-1.0.0-SNAPSHOT.jar\n",
+    "%AddJar file:/data/libs/rapids-4-spark-0.1-SNAPSHOT.jar\n",
```
Comment (GaryShen2008) [inline examples/notebooks/scala/taxi_gpu_crossvalidation.ipynb:153]: Please update these jars.

File: getting-started-guides/on-prem-cluster/standalone-python.md [hunk 9]
```diff
@@ -252,13 +272,16 @@ ${SPARK_HOME}/bin/spark-submit
  --py-files ${SPARK_PY_FILES}                                                   \
  ${SPARK_PYTHON_ENTRYPOINT}                                                     \
  --mainClass=${EXAMPLE_CLASS}                                                   \
- --dataPath=train::${DATA_PATH}/mortgage/csv/train/mortgage_train_merged.csv      \
- --dataPath=trans::${DATA_PATH}/mortgage/csv/test/mortgage_eval_merged.csv         \
+ --dataPath=train::${DATA_PATH}/mortgage/out/train/      \
+ --dataPath=trans::${DATA_PATH}/mortgage/out/eval/         \
  --format=csv                                                                   \
```
Comment (GaryShen2008) [inline getting-started-guides/on-prem-cluster/standalone-python.md]: Should the format be parquet?

File: datasets/ETL/MortgageETL.ipynb (hunk 1 above)
Comment (firestarman) [reply datasets/ETL/MortgageETL.ipynb]: updated

File: datasets/ETL/MortgageETL.ipynb (hunk 2 above)
Comment (firestarman) [reply datasets/ETL/MortgageETL.ipynb]: updated

File: examples/apps/scala/pom.xml (hunk 3 above)
Comment (firestarman) [reply examples/apps/scala/pom.xml]: updated

File: getting-started-guides/on-prem-cluster/yarn-python.md [hunk 10]
```diff
@@ -225,8 +249,8 @@ ${SPARK_HOME}/bin/spark-submit
  --py-files ${SPARK_PY_FILES}                                                   \
  ${SPARK_PYTHON_ENTRYPOINT}                                                     \
  --mainClass=${EXAMPLE_CLASS}                                                   \
- --dataPath=train::${DATA_PATH}/mortgage/csv/train/mortgage_train_merged.csv      \
- --dataPath=trans::${DATA_PATH}/mortgage/csv/test/mortgage_eval_merged.csv         \
+ --dataPath=train::${DATA_PATH}/mortgage/out/train/       \
+ --dataPath=trans::${DATA_PATH}/mortgage/out/eval/         \
  --format=csv                                                                   \
```
Comment (GaryShen2008) [inline getting-started-guides/on-prem-cluster/yarn-python.md]: Should the format be parquet?

**Review by GaryShen2008 — COMMENTED** [review]
Need to fix the comments

File: examples/notebooks/python/agaricus-gpu.ipynb (hunk 4 above)
Comment (firestarman) [reply examples/notebooks/python/agaricus-gpu.ipynb]: restored

File: examples/notebooks/python/mortgage-gpu.ipynb (hunk 5 above)
Comment (firestarman) [reply examples/notebooks/python/mortgage-gpu.ipynb]: restored

File: getting-started-guides/on-prem-cluster/yarn-python.md (hunk 10 above)
Comment (firestarman) [reply getting-started-guides/on-prem-cluster/yarn-python.md]: @shotai please help confirm.

File: getting-started-guides/on-prem-cluster/standalone-python.md (hunk 9 above)
Comment (firestarman) [reply getting-started-guides/on-prem-cluster/standalone-python.md]: @shotai please help confirm.

File: examples/notebooks/scala/taxi_gpu_crossvalidation.ipynb (hunk 8 above)
Comment (firestarman) [reply examples/notebooks/scala/taxi_gpu_crossvalidation.ipynb:153]: Let's wait for the feedback from @wbo4958 for the whole file's status.

File: getting-started-guides/on-prem-cluster/yarn-python.md (hunk 10 above)
Comment (firestarman) [reply getting-started-guides/on-prem-cluster/yarn-python.md]: updated after profiling the whole process

File: getting-started-guides/on-prem-cluster/standalone-python.md (hunk 9 above)
Comment (firestarman) [reply getting-started-guides/on-prem-cluster/standalone-python.md]: updated after profiling the whole process

**Review by GaryShen2008 — APPROVED** [review]

File: examples/notebooks/scala/mortgage_gpu_crossvalidation.ipynb (hunk 6 above)
Comment (GaryShen2008) [reply examples/notebooks/scala/mortgage_gpu_crossvalidation.ipynb]: Will fix later.

File: examples/notebooks/scala/taxi_gpu_crossvalidation.ipynb (hunk 7 above)
Comment (GaryShen2008) [reply examples/notebooks/scala/taxi_gpu_crossvalidation.ipynb:32]: Will fix later.

**Pull request merged**